include/gtirb/Allocator.hpp

```diff
@@ -264,6 +264,44 @@ class BumpPtrAllocatorImpl {
   }
 };
 
+/// Allocate memory for a specific type in an ever-growing pool, and call
+/// destructors for all allocated objects when freeing the pool.
+template <class T, size_t SlabSize = 4096, size_t SizeThreshold = SlabSize>
+class DtorBumpPtrAllocator : BumpPtrAllocatorImpl<SlabSize, SizeThreshold> {
```

Comment:
The way that I'm used to seeing this done is more like `SpecificBumpPtrAllocator` (https://llvm.org/viewvc/llvm-project/llvm/trunk/include/llvm/Support/Allocator.h?view=markup#l415) where it composes rather than inherits. Did you consider such a design?

Reply:
Oh, I wish I'd know that existed. I should probably throw out my allocator and use that instead, since it goes along with the `BumpPtrAllocator` that we've already copied.